cmd/kyma/alpha/uninstall/cmd.go

```diff
@@ -57,30 +64,63 @@ func (cmd *command) Run() error {
 		return errors.Wrap(err, "Could not initialize the Kubernetes client. Make sure your kubeconfig is valid")
 	}
 
-	var resourcePath = filepath.Join(cmd.opts.WorkspacePath, "kyma", "resources")
-	installationCfg := installConfig.Config{
+	var ui asyncui.AsyncUI
+	if !cmd.Verbose { //use async UI only if not in verbose mode
+		ui = asyncui.AsyncUI{StepFactory: &cmd.Factory}
+		if err := ui.Start(); err != nil {
+			return err
+		}
+		defer ui.Stop()
+	}
+
+	//initialize deploy step factory
+	stepFactory = deploy.NewUIStepFactory(cmd.Verbose, ui)
+
+	// retrieve Kyma metadata (provides details about the current Kyma installation)
+	kymaMeta, err := cmd.retrieveKymaMetadata()
+	if err != nil {
+		return err
+	}
+
+	var resourcePath = filepath.Join(cmd.opts.WorkspacePath, "resources")
+	installCfg := installConfig.Config{
 		WorkersCount:                  cmd.opts.WorkersCount,
 		CancelTimeout:                 cmd.opts.CancelTimeout,
 		QuitTimeout:                   cmd.opts.QuitTimeout,
 		HelmTimeoutSeconds:            int(cmd.opts.HelmTimeout.Seconds()),
 		BackoffInitialIntervalSeconds: 3,
 		BackoffMaxElapsedTimeSeconds:  60 * 5,
 		Log:                           cli.LogFunc(cmd.Verbose),
-		ComponentsListFile:            filepath.Join(cmd.opts.WorkspacePath, "kyma", "installation", "resources", "components.yaml"),
+		ComponentsListFile:            filepath.Join(cmd.opts.WorkspacePath, "installation", "resources", fmt.Sprintf("uninstall-%s", kymaMeta.ComponentListFile)),
 		CrdPath:                       filepath.Join(resourcePath, "cluster-essentials", "files"),
 		ResourcePath:                  resourcePath,
-		Version:                       "-",
+		Version:                       kymaMeta.Version,
 	}
 
-	var ui asyncui.AsyncUI
-	if cmd.Verbose {
-		defer log.Println("Kyma uninstalled!")
-	} else {
-		asyncUI := asyncui.AsyncUI{StepFactory: &cmd.Factory}
-		if err = asyncUI.Start(); err != nil {
+	// only download if not from local sources
+	if kymaMeta.Version != localSource { //KymaMetadata.Version contains the "source" value which was used during the installation
+		//if workspace already exists ask user for deletion-approval
+		_, err := os.Stat(cmd.opts.WorkspacePath)
+		approvalRequired := !os.IsNotExist(err)
+
+		if err := deploy.CloneSources(stepFactory, cmd.opts.WorkspacePath, kymaMeta.Version); err != nil {
 			return err
 		}
-		defer asyncUI.Stop() // stop receiving update-events and wait until UI rendering is finished
+		// delete workspace folder
+		if approvalRequired {
+			userApprovalStep := stepFactory.AddStep("Workspace folder exists")
+			if userApprovalStep.PromptYesNo(fmt.Sprintf("Delete workspace folder '%s' after Kyma was removed?", cmd.opts.WorkspacePath)) {
```

Comment:
We shouldn't prompt in case `non-interactive` or `ci` flags are passed.

Reply:
right.. let me add that

Reply:
fixed